internal/eventbus/event_bus.go

```diff
@@ -51,7 +51,16 @@ func (b *EventBus) NumClientSubscriptions(clientID string) int {
 }
 
 func (b *EventBus) SubscribeWithArgs(ctx context.Context, args tmpubsub.SubscribeArgs) (Subscription, error) {
-	return b.pubsub.SubscribeWithArgs(ctx, args)
+	sub, err := b.pubsub.SubscribeWithArgs(ctx, args)
+	if sub != nil {
+		return sub, err
+	}
+
+	// A nil-valued *tmpubsub.Subscription is still a valid implementation of
+	// Subscription. Internally, interface values are `(concrete-type, value)`.
+	// In this case, it would be `(*tmpubsub.Subscription, nil)`. That will
+	// cause panics, so instead we need to explicitly return nil.
+	return nil, err
```

Comment:
The only time the pubsub `SubscribeWithArgs` can return a nil value is if there is an error. When a function reports an error, the caller should treat the non-error values as garbage, unless the function explicitly documents otherwise (e.g., `io.Reader`). 

It can't hurt to be defensive here, but if a caller is checking the `sub` value before verifying that there was no error, the caller is buggy and should be fixed.

```suggestion
	sub, err := b.pubsub.SubscribeWithArgs(ctx, args)
	if err != nil {
		return nil, err
	}
	return sub, nil
```